### PR TITLE
Issue search related tests

### LINKED ISSUE #997: [Sveltekit-pod] Implement unit test for all components
As an user I would like all the components in the components folder to have unit tests. 

NOTE: This ticket is going to be worked on from multiple people and will have multiple PRs

AC: (add you name next to the component that you are working on)
- [ ] Example component -  Simone
- [x] Error Page - Ian
- [x] FileExplorer - Allan
- [x] FileViewer - Ian
- [x] Gists - Ian
- [x] IssueSearch - Allan
- [x] NavBar - Ian
- [x] PRHeader - Ian
- [x] PRIssueCard - Ian
- [x] PRsList - Ian
- [x] Profile - Ian
- [x] RepoHeader - Ian
- [ ] RepoList - Ian
- [x] RepositoryCard - Ian
- [x] TopRepositories - Ian
- [ ] auth - Ian
- [ ] shared

### PR DESCRIPTION
**Pull request by AllanJeremy**
Partly fixes #997

### CODE REVIEW

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      }
+    );
+
+    it('should contain an issue number', () => {
+      const issueNumbers = screen
+        .queryAllByTestId('issue-number')
+        .map((number) => number.textContent)
+        .filter(Boolean);
+
+      expect(issueNumbers.length).toBe(4);
```
Comment (mbicknese) [inline svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:34]: It doesn't really hurt, but you're checking for the length to be four only to follow it up by checking whether the variable equals a 4-length array. You could drop this line and keep the same level of confidence in the code.

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      expect(issueNumbers).toEqual(['#1', '#2', '#3', '#4']);
+    });
+
+    it('should contain the comments count', () => {
+      const commentCounts = screen
+        .queryAllByTestId('issue-comments-count')
+        .map((comment) => comment.textContent)
+        .filter(Boolean);
+
+      expect(commentCounts.length).toBe(4);
```
Comment (mbicknese) [inline svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:44]: Same as the previous comment.

**Review by iansamz — APPROVED** [review]

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  beforeEach(() => {
+    render(IssueSearchList, {
+      items: MOCK_ISSUE_ARRAY,
+    });
+  });
+
+  describe('List issues', () => {
+    it('should list all issues', () => {
+      const issues = screen.queryAllByTestId('issue-search-list-item');
+      expect(issues.length).toBe(4);
```
Comment (mbicknese) [inline svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:16]: We could extract the 4 and make it a const at the top of the file. Maybe even base it on the length of mock issues. This way the tests won't fail whenever the number of issues change.

**Review by mbicknese — CHANGES_REQUESTED** [review]
Two optimization points. Also there's linting warning (see GitHub comments) you might want to look at.

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts (hunk 3 above)
Comment (AllanJeremy) [reply svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:16]: Makes sense!

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts (hunk 1 above)
Comment (AllanJeremy) [reply svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:34]: True

Comment (AllanJeremy) [reply svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts:34]: In the second case we are testing for the order in which stuff is displayed though @mbicknese

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts (hunk 3 above)
Comment (AllanJeremy) [reply svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts]: Fixed

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts (hunk 2 above)
Comment (AllanJeremy) [reply svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts]: fixed

File: svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts (hunk 1 above)
Comment (AllanJeremy) [reply svelte-kit-scss/src/lib/components/IssueSearch/IssueSearch.spec.ts]: Fixed

**Comment by AllanJeremy** [comment]
Fixed all mentioned issues. Linting errors existed before PR, fixed one (unused var) but left the other two accessibility errors (can't use `#`) since I don't as yet know if they would alter any behaviour

**Review by mbicknese — APPROVED** [review]
⛵

**Pull request merged**